jax/lax_linalg.py

```diff
@@ -76,11 +79,10 @@ def cholesky_jvp_rule(primals, tangents):
   L = cholesky_p.bind(x)
 
   # Forward-mode rule from https://arxiv.org/pdf/1602.07527.pdf
-  sigma_dot = (sigma_dot + _T(sigma_dot)) / 2
   phi = lambda X: np.tril(X) / (1 + np.eye(x.shape[-1]))
   tmp = triangular_solve(L, sigma_dot,
                          left_side=False, transpose_a=True, lower=True)
-  L_dot = lax.dot(L, phi(triangular_solve(
+  L_dot = np.matmul(L, phi(triangular_solve(
```

Comment:
I'm noting some type mismatches in the test failures. It's possible we need a dtype= argument here or something like that.

Reply:
I think we can use `lax.batch_matmul` here instead to avoid NumPy's dtype promotion.